packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -29,71 +29,124 @@ import {
 	BlockControls,
 	InnerBlocks,
 	InspectorControls,
-	URLPopover,
+	__experimentalLinkControl as LinkControl,
 } from '@wordpress/block-editor';
-import {
-	Fragment,
-	useRef,
-	useState,
-} from '@wordpress/element';
+import { Fragment, useState, useRef, useEffect } from '@wordpress/element';
 
 function NavigationMenuItemEdit( {
 	attributes,
 	isSelected,
 	isParentOfSelectedBlock,
 	setAttributes,
+	fetchSearchSuggestions,
 } ) {
-	const plainTextRef = useRef( null );
+	const { label, opensInNewTab, title, url } = attributes;
+	const link = title ? { title, url } : null;
 	const [ isLinkOpen, setIsLinkOpen ] = useState( false );
-	const [ isEditingLink, setIsEditingLink ] = useState( false );
-	const [ urlInput, setUrlInput ] = useState( null );
+	const [ wasCloseByLinkControl, setWasCloseByLinkControl ] = useState( false );
+
+	const plainTextRef = useRef( null );
+
+	/**
+	 * It's a kind of hack to handle closing the LinkControl popover
+	 * clicking on the ToolbarButton link.
+	 */
+	useEffect( () => {
+		if ( ! isSelected ) {
+			setIsLinkOpen( false );
+			setWasCloseByLinkControl( false );
+		}
+		return () => {
+			setIsLinkOpen( false );
+			setWasCloseByLinkControl( false );
+		};
+	}, [ isSelected ] );
 
-	const inputValue = urlInput !== null ? urlInput : url;
+	/**
+	 * `onKeyDown` LinkControl handler.
+	 * It takes over to stop the event propagation to make the
+	 * navigation work, avoiding undesired behaviors.
+	 * For instance, it will block to move between menu items
+	 * when the LinkOver is focused.
+	 *
+	 * @param {Event} event
+	 */
+	const handleLinkControlOnKeyDown = ( event ) => {
+		const { keyCode } = event;
 
-	const onKeyDown = ( event ) => {
-		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( keyCode ) > -1 ) {
 			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
 			event.stopPropagation();
 		}
 	};
 
-	const closeURLPopover = () => {
-		setIsEditingLink( false );
-		setUrlInput( null );
-		setIsLinkOpen( false );
-	};
-
-	const autocompleteRef = useRef( null );
-
-	const onFocusOutside = ( event ) => {
-		const autocompleteElement = autocompleteRef.current;
-		if ( autocompleteElement && autocompleteElement.contains( event.target ) ) {
-			return;
+	/**
+	 * Updates the link attribute when it changes
+	 * through of the `onLinkChange` LinkControl callback.
+	 *
+	 * @param {Object|null} itemLink Link object if it has been selected. Otherwise, Null.
+	 */
+	const updateLink = ( itemLink ) => {
+		if ( ! itemLink ) {
+			// If not link, then set empty string to link attrs.
+			return setAttributes( { title: '', url: '' } );
```

Comment:
Just wondering. Could we use destructing and default values to avoid having the conditional?

Would this work?

```js
const updateLink = ( { title: '', url: '' } ) => {
		setAttributes( {title, url} );
	};
```

Reply:
I've addressed this. Was a little more complex than I wrote out in pseudo code.